src/main/java/picard/vcf/FixVcfHeader.java

```diff
@@ -121,41 +114,54 @@ public static void main(final String[] args) {
         if (HEADER != null) IOUtil.assertFileIsReadable(HEADER);
         IOUtil.assertFileIsWritable(OUTPUT);
 
-        final VCFFileReader reader = new VCFFileReader(INPUT, false);
+        final VCFFileReader reader     = new VCFFileReader(INPUT, false);
+        final VCFHeader existingHeader = reader.getFileHeader();
 
         final VCFHeader outHeader;
         if (HEADER != null) { // read the header from the file
             final VCFFileReader headerReader = new VCFFileReader(HEADER, false);
+            final VCFHeader inputHeader      = headerReader.getFileHeader();
             if (ENFORCE_SAME_SAMPLES) {
-                outHeader = headerReader.getFileHeader();
-                enforceSameSamples(reader.getFileHeader(), outHeader);
+                outHeader = inputHeader;
+                enforceSameSamples(existingHeader, outHeader);
             }
             else {
-                outHeader = new VCFHeader(headerReader.getFileHeader().getMetaDataInInputOrder(), reader.getFileHeader().getSampleNamesInOrder());
+                outHeader = new VCFHeader(inputHeader.getMetaDataInInputOrder(), existingHeader.getSampleNamesInOrder());
             }
             CloserUtil.close(headerReader);
+            outHeader.getFilterLines()
+                    .stream()
+                    .filter(filterHeaderLine -> !existingHeader.hasFilterLine(filterHeaderLine.getID()))
+                    .forEach(filterHeaderLine -> log.info("FILTER line found in HEADER will be added to OUTPUT: " + filterHeaderLine));
             outHeader.getInfoHeaderLines()
                     .stream()
-                    .filter(infoHeaderLine -> !reader.getFileHeader().hasInfoLine(infoHeaderLine.getID()))
+                    .filter(infoHeaderLine -> !existingHeader.hasInfoLine(infoHeaderLine.getID()))
                     .forEach(infoHeaderLine -> log.info("INFO line found in HEADER will be added to OUTPUT: " + infoHeaderLine.getID()));
```

Comment:
It would be slightly more streamlined to convert the stream to ID, e.g.
```
outHeader.getInfoHeaderLines()
        .stream()
        .map(VCFCompoundHeaderLine::getID)
        .filter(id -> !reader.getFileHeader().hasInfoLine(id))
        .forEach(id -> log.info("INFO line found in HEADER will be added to OUTPUT: " + id));
```

Reply:
Done, but I decided to use `existingHeader` instead of `reader.getFileHeader`, since I have it on hand.